### ISSUE TITLE
Build Fail on 1.14.0 2016-10-02

### ISSUE THREAD

**Issue by KodrAus** [serde-rs/aster#113]
I'm getting a build error on the latesr nightly for `aster`:

```
error[E0063]: missing field `attrs` in initializer of `syntax::ast::LifetimeDef`
  --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.27.0/src/lifetime.rs:43:9
   |
43 |         ast::LifetimeDef {
   |         ^^^^^^^^^^^^^^^^ missing `attrs`
error[E0063]: missing field `attrs` in initializer of `syntax::ast::LifetimeDef`
   --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.27.0/src/lifetime.rs:111:30
    |
111 |         self.callback.invoke(ast::LifetimeDef {
    |                              ^^^^^^^^^^^^^^^^ missing `attrs`
error: no associated item named `Vec` found for type `syntax::ast::TyKind` in the current scope
   --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.27.0/src/ty.rs:142:28
    |
142 |         self.build_ty_kind(ast::TyKind::Vec(ty))
    |                            ^^^^^^^^^^^^^^^^
error[E0063]: missing field `attrs` in initializer of `syntax::ast::TyParam`
   --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.27.0/src/ty_param.rs:103:30
    |
103 |         self.callback.invoke(ast::TyParam {
    |                              ^^^^^^^^^^^^ missing `attrs`
error: aborting due to 4 previous errors
```

**Comment by zmarcantel** [comment]
Same thing for `1.14.0-nightly (9c31d76e9 2016-10-03)` (with one additional error).

```
error[E0063]: missing field `attrs` in initializer of `syntax::ast::LifetimeDef`
  --> /home/zach/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.29.0/src/lifetime.rs:43:9
   |
43 |         ast::LifetimeDef {
   |         ^^^^^^^^^^^^^^^^ missing `attrs`

error[E0063]: missing field `attrs` in initializer of `syntax::ast::LifetimeDef`
   --> /home/zach/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.29.0/src/lifetime.rs:111:30
    |
111 |         self.callback.invoke(ast::LifetimeDef {
    |                              ^^^^^^^^^^^^^^^^ missing `attrs`

error: no associated item named `Vec` found for type `syntax::ast::TyKind` in the current scope
   --> /home/zach/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.29.0/src/ty.rs:148:28
    |
148 |         self.build_ty_kind(ast::TyKind::Vec(ty))
    |                            ^^^^^^^^^^^^^^^^

error: no associated item named `FixedLengthVec` found for type `syntax::ast::TyKind` in the current scope
   --> /home/zach/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.29.0/src/ty.rs:153:28
    |
153 |         self.build_ty_kind(ast::TyKind::FixedLengthVec(ty, len_expr))
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0063]: missing field `attrs` in initializer of `syntax::ast::TyParam`
   --> /home/zach/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.29.0/src/ty_param.rs:103:30
    |
103 |         self.callback.invoke(ast::TyParam {
    |                              ^^^^^^^^^^^^ missing `attrs`
```

Was going to make a new issue, but the failures are so similar and only 1 nightly release apart so decided against it.

The only difference between the original issue and this on is:

```
error: no associated item named `FixedLengthVec` found for type `syntax::ast::TyKind` in the current scope
   --> /home/zach/.cargo/registry/src/github.com-1ecc6299db9ec823/aster-0.29.0/src/ty.rs:153:28
    |
153 |         self.build_ty_kind(ast::TyKind::FixedLengthVec(ty, len_expr))
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by dashed** [comment]
Note for issue lurkers: I rolled back nightly build to `rustc 1.14.0-nightly (86affcdf6 2016-09-28)` and this error is gone.

**Comment by freiguy1** [comment]
I ran into this issue last night as well. I was attempting to update lib versions and nightly version for a web api I'm building which uses diesel and serde. Apparently the newest version of diesel needs rust nightly >= 2016-10-10, and aster requires nightly <= 2016-09-28. Sad day.  I don't envy the person attempting to get a new project up and running with both serde and diesel.  At least I have a set of versions which work together from 2016-03-25!

**Comment by oli-obk** [comment]
Note: it is suggested to use serde_derive instead of serde codegen, since serde_derive is expected to never break and will become available on stable in 3 monts or so. Afaik diesel is moving in a similar direction

**Comment by dtolnay** [comment]
Fixed in #115.
